### PR TITLE
4270 get the Archive URLs for imported works

### PR DESCRIPTION
**Pull request by ariana-paris**
https://code.google.com/p/otwarchive/issues/detail?id=4270

This creates an end point which takes an array of external urls and for each one, returns either an error indicating it hasn't been found, or the work's URL and creation date on the Archive.

### CODE REVIEW

**Comment by sarken** [comment]
@shalott Hi! Do you have a moment to look at this, since you reviewed the other API pull request?

File: app/controllers/api/v1/works_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  # Params:
+  # +original_urls+:: an array of original URLs to find on the Archive
+  def batch_urls
+    original_urls = params[:original_urls]
+    status = :bad_request
+    results = []
+    if original_urls.nil? || original_urls.blank? || original_urls.size == 0
+      results << { error: "Please provide a list of URLs to find." }
+    elsif original_urls.size >= ArchiveConfig.IMPORT_MAX_CHAPTERS
+      results << { error: "Please provide no more than #{ ArchiveConfig.IMPORT_MAX_CHAPTERS } URLs to find" }
```
Comment (sarken) [inline app/controllers/api/v1/works_controller.rb]: My tiny useless contribution: This error message doesn't have a period at the end, but the others do.

Comment (ariana-paris) [inline app/controllers/api/v1/works_controller.rb]: Damn. I thought I was being so careful, too. *fixes*

File: app/controllers/api/v1/works_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+class Api::V1::WorksController < Api::V1::BaseController
+  respond_to :json
+
+  # Return the URLs of a batch of individual works. Limits the number of URLs to
+  # IMPORT_MAX_CHAPTERS so it doesn't get tied up in checking URLs for too long.
+  # Params:
+  # +original_urls+:: an array of original URLs to find on the Archive
+  def batch_urls
```
Comment (shalott) [inline app/controllers/api/v1/works_controller.rb]: maybe move lines 18-31 into a separate method process_batch_urls eg?

File: app/controllers/api/v1/import_controller.rb [hunk 3]
```diff
@@ -67,7 +71,7 @@ def import_work(archivist, external_work)
         @some_success = true
         work_status = :created
         work_url = work_url(work)
-        work_messages << "Successfully created work \"" + work.title + "\""
+        work_messages << "Successfully created work \"" + work.title + "\"."
```
Comment (shalott) [inline app/controllers/api/v1/import_controller.rb]: haha I approve <3

**Comment by shalott** [comment]
Looks good to me.

**Comment by sarken** [comment]
Thanks for reviewing!

Marking needs coder action for the comments on lines 18-21, which are for whatever reason not showing up on the conversation page.

File: app/controllers/api/v1/works_controller.rb (hunk 2 above)
Comment (ariana-paris) [inline app/controllers/api/v1/works_controller.rb]: I've given it a try - let's see if Hounds likes it better like that!

Comment (ariana-paris) [inline app/controllers/api/v1/works_controller.rb]: Well, it lowered the score but didn't get rid of the message! I think I'm going to leave it at that...

**Pull request merged**